### DISCUSSION TITLE
Getting bundle configuration in another compiler pass?

### DISCUSSION THREAD

**Discussion by pkly** [doctrine/DoctrineBundle#1543] [Q&A]
Hi, I recently wrote a bundle for one of our applications, and I needed to find the doctrine repositories.
The services themselves are easy to find, since they're tagged, but the entity they're responsible for is not known easily by just looking at the service definition in the container.

Is it possible to somehow learn of what an entity is being provided by a repository service with the definition alone inside of a compiler pass?

Something like
```php
// in CP
foreach ($container->findTaggedServiceIds('doctrine.orm_repository') as $id => $tags) {
    $definition = $container->getDefinition($id);
    $entityClass = ....?
}
```

In theory it should be possible by looking at the second parameter for each definition, right? But since one could simply use a custom repository I was wondering if there's another way.

**Comment by dmaicher** [comment]
Maybe this information is somewhere in the entity metadata? There should be a `Doctrine\ORM\Configuration` instance per configured entity manager.

**Reply by pkly** [reply]
I'll check that and get back to you, thanks

**Reply by pkly** [reply]
Hi! It took a while for me to get back to it, but I just checked it out and I think it's _something_, but the fact that it's a service instead of static metadata makes it harder for me to actually do the logic I need, since my configuration at this point needs to be done at build time...

I have an idea on how to solve this maybe though, but it won't be as optimized as my previous solution, since that already had id-fqcn maps when the container is created.
